### DISCUSSION TITLE
未登录跳转登录页不成功问题

### DISCUSSION THREAD

**Discussion by luwq168** [dotnetcore/BootstrapBlazor#1459] [Q&A] [answered]
依模板创建Server项目，添加Microsoft 标识，相关代码如下：
App.razor:
`<BootstrapBlazorRoot>
    <CascadingAuthenticationState>
        <Router AppAssembly="@typeof(MainLayout).Assembly">
            <Found Context="routeData">
                <AuthorizeRouteView RouteData="@routeData"
                                    DefaultLayout="@typeof(MainLayout)">
                    <NotAuthorized>
                        @if (context.User.Identity == null || !context.User.Identity.IsAuthenticated)
                        {
                            <RedirectToLogin />
                        }
                        else
                        {
                            <p role="alert">登录账户没有权限使用该功能！</p>
                        }
                    </NotAuthorized>
                </AuthorizeRouteView>
            </Found>
            <NotFound>
                <PageTitle>Not found</PageTitle>
                <LayoutView Layout="@typeof(MainLayout)">
                    <p role="alert">对不起，未找到相应页面显示！</p>
                </LayoutView>
            </NotFound>
        </Router>
    </CascadingAuthenticationState>
</BootstrapBlazorRoot>
`
RedirectToLogin.razor:
`@inject NavigationManager Navigation
@code {
    protected override void OnAfterRender(bool firstRender)
    {
        Navigation.NavigateTo("/account/login");
    }
//或
    protected override void OnInitialized()
    {
        Navigation.NavigateTo("/account/login");
    }
}
`
发现未起作用。在其他项目能正常跳转。

**Comment by ArgoZhang** [comment, marked as answer]
才看到留言，这个格式化一下让阅读者方便阅读比较好哈
